### PR TITLE
Document card updates

### PR DESCRIPTION
**Pull request by yiminwu**
Include render file list in preview area and compact card layout for small screen.

### CODE REVIEW

File: src/components/DocumentCard/DocumentCardPreview.scss [hunk 1]
```diff
@@ -1,12 +1,56 @@
...
+  @include padding(16px, 16px, 0, 16px);
+  list-style-type: none;
+  margin: 0;
+
+  li {
+    height: 16px;
+    line-height: 16px; // Vertically center the text
+    margin-bottom: 8px;
+    overflow: hidden;
+    padding-left: 32px; // Make room for icon
```
Comment (mikewheaton) [inline src/components/DocumentCard/DocumentCardPreview.scss]: Localization: `@include padding-left: 32px;`

File: src/components/DocumentCard/DocumentCardPreview.scss [hunk 2]
```diff
@@ -1,12 +1,56 @@
...
+    color: $ms-color-neutralDark;
+
+    &:hover {
+      color: $ms-color-themePrimary;
+    }
+  }
+}
+
+.ms-DocumentCardPreview-fileListIcon {
+  left: 0;
```
Comment (mikewheaton) [inline src/components/DocumentCard/DocumentCardPreview.scss]: Localization: `@include left(0);`

File: src/components/DocumentCard/DocumentCard.Props.ts [hunk 3]
```diff
@@ -22,20 +28,53 @@ export interface IDocumentCardProps extends React.Props<DocumentCard> {
...
 
 export interface IDocumentCardPreviewProps extends React.Props<DocumentCardPreview> {
   /**
    * One or more preview images to display.
    */
   previewImages: IDocumentCardPreviewImage[];
+
+  /**
+   * The format for the text that should display the number of overflow documents.
+   * such as '+2 more'
```
Comment (dzearing) [inline src/components/DocumentCard/DocumentCard.Props.ts]: Maybe this should be the type
(count) => string
And then example comment should be:

```typescript
(displayCount) => `=${ displayCount } more`;
```

File: src/components/DocumentCard/DocumentCardPreview.tsx [hunk 4]
```diff
@@ -1,80 +1,108 @@
...
 export class DocumentCardPreview extends React.Component<IDocumentCardPreviewProps, any> {
-
-  private _async: Async;
-  private _interval: number;
-
   constructor(props: IDocumentCardPreviewProps) {
     super(props);
-    this._showNextPreview = this._showNextPreview.bind(this);
-    this._async = new Async(this);
+    this._renderPreviewList = this._renderPreviewList.bind(this);
```
Comment (dzearing) [inline src/components/DocumentCard/DocumentCardPreview.tsx]: import autobind and use @autobind

**Comment by dzearing** [comment]
Approved

**Pull request merged**